library-udf/src/test/java/org/apache/iotdb/library/string/StringTests.java

```diff
@@ -0,0 +1,208 @@
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 200, "A,A+,A,B+"));
+      statement.execute(
+          String.format("insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 300, "B+,B,B"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 400, "A+,A,A+,A"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 500, "A,B-,B,B"));
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function regexmatch as 'org.apache.iotdb.library.string.UDTFRegexMatch'");
+      statement.execute(
+          "create function regexreplace as 'org.apache.iotdb.library.string.UDTFRegexReplace'");
+      statement.execute(
+          "create function regexsplit as 'org.apache.iotdb.library.string.UDTFRegexSplit'");
+      statement.execute(
+          "create function strreplace as 'org.apache.iotdb.library.string.UDTFStrReplace'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setUdfCollectorMemoryBudgetInMB(100)
+        .setUdfTransformerMemoryBudgetInMB(100)
+        .setUdfReaderMemoryBudgetInMB(100);
+  }
+
+  @Test
+  public void testRegexMatch1() {
+    String sqlStr =
+        "select regexmatch(d1.s1,\"regex\"=\"\\d+\\.\\d+\\.\\d+\\.\\d+\", \"group\"=\"0\") from root.vehicle";
+    try (Connection connection =
+            DriverManager.getConnection(
+                Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      assert resultSetLength == 5;
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testRegexReplace1() {
+    String sqlStr =
+        "select regexreplace(d2.s1,\"regex\"=\"192\\.168\\.0\\.(\\d+)\", \"replace\"=\"cluster-$1\") from root.vehicle";
+    try (Connection connection =
+            DriverManager.getConnection(
+                Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      assert resultSetLength == 5;
```

Comment:
Use `Assert.assertEquals()`. Don't use `assert` keyword as it doesn't belong to UT framework and need special JVM parameters to work.
BTW, I think we should assert the returned columns are as our expcted. Row number assertion is insufficient.

Reply:
Done.